src/linear_algebra/finsupp.lean

```diff
@@ -313,10 +313,30 @@ finsupp.sum_single_index (f i).map_zero
 
 theorem lsum_symm_apply (f : (α →₀ M) →ₗ[R] N) (x : α) : lsum.symm f x = f.comp (lsingle x) := rfl
 
+section
+variables (M) (R) (X : Type*)
+
+/--
+A slight rearrangement from `lsum` gives us
+the bijection underlying the free-forgetful adjunction for R-modules.
+-/
+noncomputable def lift : (X → M) ≃ ((X →₀ R) →ₗ[R] M) :=
+(equiv.arrow_congr (equiv.refl _) (ring_lmap_equiv_self R M).symm).trans (lsum : _ ≃+ _).to_equiv
```

Comment:
This line will be broken by #6183.

Reply:
Aha, thanks for pointing that out.

Reply:
@eric-wieser, @jcommelin, I've now added `mul_equiv.arrow_congr`, the multiplicative version of `equiv.arrow_congr` (only makes sense when the targets are commutative), and used this to show that `lift` here is actually an `add_equiv`.